src/map/npc.c

```diff
@@ -2805,19 +2805,18 @@ static int npc_selllist_sub(struct map_session_data *sd, struct itemlist *item_l
 static int npc_selllist(struct map_session_data *sd, struct itemlist *item_list)
 {
 	int64 z;
-	int i,skill_t, skill_idx = skill->get_index(MC_OVERCHARGE);
+	int i, skill_t, skill_idx = skill->get_index(MC_OVERCHARGE);
 	struct npc_data *nd;
 	bool duplicates[MAX_INVENTORY] = { 0 };
 
 	nullpo_retr(1, sd);
 	nullpo_retr(1, item_list);
 
-	if( ( nd = npc->checknear(sd, map->id2bl(sd->npc_shopid)) ) == NULL ) {
+	if ((nd = npc->checknear(sd, map->id2bl(sd->npc_shopid))) == NULL)
 		return 1;
-	}
 
-	if( nd->subtype != SHOP ) {
-		if (!(nd->subtype == SCRIPT && nd->u.scr.shop && (nd->u.scr.shop->type == NST_ZENY || nd->u.scr.shop->type == NST_MARKET)))
+	if (nd->subtype != SHOP) {
+		if ((nd->subtype == SCRIPT && nd->u.scr.shop && (nd->u.scr.shop->type == NST_ZENY || nd->u.scr.shop->type == NST_MARKET)) == NULL)
```

Comment:
Comparing the expression's result with `NULL` is incorrect. It has to be `true` or `false`. Or use keep using the "Not"-operator (`!`).

If you want to get rid of the "Not"-operator by all means, just invert the conditions:
`if (nd->subtype != SCRIPT || nd->u.scr.shop == NULL || (nd->u.scr.shop->type != NST_ZENY && nd->u.scr.shop->type != NST_MARKET))`